### PR TITLE
fix: env handle optional values

### LINKED ISSUE #3858: Crash when setting $ENABLE_ASYNC_PROMPT=True
Seems $ASYNC_PROMPT_THREAD_WORKERS has an invalid default value.

## Repro
 ```
$ xonsh --no-rc
 . . snip . .
(x1)  $ $XONSH_SHOW_TRACEBACK = True
(x1)  $ $ENABLE_ASYNC_PROMPT=True
```
### Expect
Should work.

### Observe
Crash as noted below.  Reference to `ASYNC_PROMPT_THREAD_WORKERS` is significant.  I can avoid the crash by doing this sequence: 
```
(x1)  $ $XONSH_SHOW_TRACEBACK = True
(x1)  $ $ASYNC_PROMPT_THREAD_WORKERS = 5
(x1)  $ $ENABLE_ASYNC_PROMPT=True
```

### Traceback (if applicable)

<details>

```
Traceback (most recent call last):
  File "c:\users\bobhy\src\xonsh\xonsh\ptk_shell\shell.py", line 276, in _get_prompt_tokens
    p = self.prompt_formatter(
  File "c:\users\bobhy\src\xonsh\xonsh\ptk_shell\formatter.py", line 31, in __call__
    self.updator = PromptUpdator(self.session)
  File "c:\users\bobhy\src\xonsh\xonsh\ptk_shell\updator.py", line 144, in __init__
    self.executor = Executor()
  File "c:\users\bobhy\src\xonsh\xonsh\ptk_shell\updator.py", line 18, in __init__
    max_workers=builtins.__xonsh__.env["ASYNC_PROMPT_THREAD_WORKERS"]
  File "c:\users\bobhy\src\xonsh\xonsh\environ.py", line 1992, in __getitem__
    val = converter(val)
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
Traceback (most recent call last):
  File "C:\Users\bobhy\venv\x1\Scripts\xonsh-script.py", line 33, in <module>
    sys.exit(load_entry_point('xonsh', 'console_scripts', 'xonsh')())
  File "c:\users\bobhy\src\xonsh\xonsh\main.py", line 422, in main
    _failback_to_other_shells(args, err)
  File "c:\users\bobhy\src\xonsh\xonsh\main.py", line 391, in _failback_to_other_shells
    raise err
  File "c:\users\bobhy\src\xonsh\xonsh\main.py", line 420, in main
    sys.exit(main_xonsh(args))
  File "c:\users\bobhy\src\xonsh\xonsh\main.py", line 458, in main_xonsh
    shell.shell.cmdloop()
  File "c:\users\bobhy\src\xonsh\xonsh\ptk_shell\shell.py", line 254, in cmdloop
    line = self.singleline(auto_suggest=auto_suggest)
  File "c:\users\bobhy\src\xonsh\xonsh\ptk_shell\shell.py", line 222, in singleline
    line = self.prompter.prompt(**prompt_args)
  File "c:\users\bobhy\venv\x1\lib\site-packages\prompt_toolkit\shortcuts\prompt.py", line 1013, in prompt
    return self.app.run(set_exception_handler=set_exception_handler)
  File "c:\users\bobhy\venv\x1\lib\site-packages\prompt_toolkit\application\application.py", line 814, in run
    return loop.run_until_complete(
  File "C:\Users\bobhy\AppData\Local\Programs\Python\Python38\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "c:\users\bobhy\venv\x1\lib\site-packages\prompt_toolkit\application\application.py", line 781, in run_async
    return await _run_async2()
  File "c:\users\bobhy\venv\x1\lib\site-packages\prompt_toolkit\application\application.py", line 763, in _run_async2
    result = await _run_async()
  File "c:\users\bobhy\venv\x1\lib\site-packages\prompt_toolkit\application\application.py", line 644, in _run_async
    self._pre_run(pre_run)
  File "c:\users\bobhy\venv\x1\lib\site-packages\prompt_toolkit\application\application.py", line 602, in _pre_run
    c()
  File "c:\users\bobhy\venv\x1\lib\site-packages\prompt_toolkit\shortcuts\prompt.py", line 1202, in pre_run2
    pre_run()
  File "c:\users\bobhy\src\xonsh\xonsh\ptk_shell\formatter.py", line 63, in start_update
    self.updator.start()
AttributeError: 'PTKPromptFormatter' object has no attribute 'updator'
```
</details>

## xonfig

<details>

```
+------------------+---------------------+
| xonsh            | 0.9.22              |
| Git SHA          | 7a55a805            |
| Commit Date      | Oct 7 14:06:16 2020 |
| Python           | 3.8.6               |
| PLY              | 3.11                |
| have readline    | False               |
| prompt toolkit   | 3.0.7               |
| shell type       | prompt_toolkit      |
| pygments         | 2.7.1               |
| on posix         | False               |
| on linux         | False               |
| on darwin        | False               |
| on windows       | True                |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
| on jupyter       | False               |
| jupyter kernel   | None                |
+------------------+---------------------+
```

</details>

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### PR DESCRIPTION
**Pull request by jnoortheen**
handle None value for ASYNC_PROMPT_THREAD_WORKERS

fixes #3858

### CODE REVIEW

File: xonsh/tools.py [hunk 1]
```diff
@@ -1358,6 +1359,11 @@ def to_itself(x):
     return x
 
 
+def to_int_or_none(x) -> tp.Optional[int]:
+    """Convert the given value to integer if possible. Otherwise return None"""
+    return int(x) if str(x).isdigit() else None
```
Comment (scopatz) [inline xonsh/tools.py:1364]: Maybe this should be `None if x is None else int(x)`

Comment (jnoortheen) [reply xonsh/tools.py:1364]: is it possible that the value to be "None" as it is retrieved as str (As I understand)?

Comment (scopatz) [reply xonsh/tools.py:1364]: Oh yeah, it is.... In that case you should check if `x.lower() == "none"`.  Other string values should still result in an error if they are not ints

Comment (jnoortheen) [reply xonsh/tools.py:1364]: Ok

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**